src/os/ui/file/kml/abstractkmlexporter.js

```diff
@@ -616,6 +616,15 @@ os.ui.file.kml.AbstractKMLExporter.prototype.processPlacemark = function(element
       });
       os.xml.appendElementNS('value', this.kmlNS, rotDataEl, String(this.getField(item, rotationColumn)));
     }
+
+    // Add line dash info as custom element since kml doesn't support it
+    var lineDash = this.getLineDash(item);
+    if (lineDash && Array.isArray(lineDash) && lineDash.length) {
+      var lineDashDataEl = os.xml.appendElementNS('Data', this.kmlNS, ed, undefined, {
+        'name': 'lineDash'
+      });
+      os.xml.appendElementNS('value', this.kmlNS, lineDashDataEl, JSON.stringify(lineDash));
+    }
```

Comment:
This may cause a merge conflict. I am adding an API for handling our internal fields that contain JSON data, and this should be incorporated into that once both are merged.